csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -212,6 +212,96 @@ private void CountSelfAndChildrenNodes(IArrowType type, ref int count)
             count++;
         }
 
+        private protected void WriteRecordBatchInternal(RecordBatch recordBatch)
```

Comment:
What about supporting `sync` write methods on `ArrowFileWriter`? Is that something left undone? Since that class inherits from this class, it is going to be a bit odd that some of its methods support sync, and some don't.

Reply:
Added the `sync` version of `WriteRecordBatch` to `ArrowFileWriter`